include/openthread/srp_server.h

```diff
@@ -308,6 +308,39 @@ otError otSrpServerSetAnycastModeSequenceNumber(otInstance *aInstance, uint8_t a
  */
 void otSrpServerSetEnabled(otInstance *aInstance, bool aEnabled);
 
+/**
+ * This function enables/disables the auto-enable mode on SRP server.
+ *
+ * This function requires `OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE` feature.
+ *
+ * When this mode is enabled, the Border Routing Manager controls if/when to enable or disable the SRP server.
+ * SRP sever is auto-enabled when Border Routing is started and only after BR is done with the initial prefix and
+ * route configurations (when the OMR and on-link prefixes are determined, advertised in emitted Router Advertisement
+ * message on infrastructure side and published in the Thread Network Data). The SRP server is auto-disabled when BR
+ * is stopped (e.g., if the infrastructure network interface is brought down or if BR gets detached).
+ *
+ * This mode can be disabled by a `otSrpServerSetAutoEnableMode()` call with @p aEnabled set to `false` or if the SRP
```

Comment:
Document the expected behavior when auto-enable mode is enabled after the Border Routing Manager has already advertising prefix/route configurations?

Reply:
Thanks. 👍 We also need to handle this situation.

Added new mechanism in new push now (i.e. if we enable "auto-enable mode" when `RoutingManager` is already up and running and done with its initial policy evaluation, we then start SRP server immediately). Also updated the unit test to cover this situation (and other combinations).

Reply:
> We also need to handle this situation.

I figured you would catch that when updating the docs :)